test/pubsub/go2go.js

```diff
@@ -0,0 +1,46 @@
+/* eslint-env mocha */
+'use strict'
+
+const chai = require('chai')
+chai.use(require('dirty-chai'))
+chai.use(require('chai-bytes'))
+const expect = chai.expect
+
+const spawnDaemons = require('../utils/spawnDaemons')
+
+describe('pubsub', () => {
+  let daemons
+
+  // Start daemons
+  before(async function () {
+    this.timeout(20 * 1000)
+
+    daemons = await spawnDaemons(2, 'go', { pubsub: true })
+
+    // connect them
+    const identify0 = await daemons[0].client.identify()
+    await daemons[1].client.connect(identify0.peerId, identify0.addrs)
+  })
+
+  // Stop daemons
+  after(async function () {
+    await Promise.all(
+      daemons.map((daemon) => daemon.stop())
+    )
+  })
+
+  it('go publish to go subscriber', async function () {
+    this.timeout(10 * 1000)
+
+    const topic = 'test-topic'
+    const data = Buffer.from('test-data')
+
+    const subscribeIterator = daemons[1].client.pubsub.subscribe(topic)
```

Comment:
```suggestion
    const subscribeIterator = await daemons[1].client.pubsub.subscribe(topic)
```

This was previously an iterator, but I refactored this to return an iterator instead. This way, you need to await for the response, as `pubsub.subscribe(topic)` will wait for the subscribe successful message from the daemon before returning the iterator